### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by Thomas-Galtier**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,23 +24,43 @@ The objective is to compare the contributivity figures obtained with the differe
...
+    * Overlap of respective datasets: distinct (by stratifying MNIST figures) vs. overlapping (with a randomized split)
+    * Size of respective datasets: equivalent vs. different
+    * Number of data partners: 3 databases A, B, C is our default scenario, but this is to be parameterized
+* ML algorithm: CNN adapted to MNIST, not too deep so it can run on CPU
+* Distributed learning approach: federated learning (other approaches to be tested in future improvements of this experiment)
+* Contributivity evaluation approach:
+    * [done] Performance scores of models trained independently on each node
+    * [futur prospect] Data Valuation by Reinforcment Learning (DVRL) 
+    * [done] Shapley values :
+     These indicators seem to be very good candidates to measure the contributivity of each data providers, because they are usually used in game theory to fairly attributes the gain of a coalitional game amongst its players, which exactly want we are looking for here.<br/><br/>
```
Comment (natct10) [inline README.md:38]: typo: which IS exactly

File: README.md [hunk 2]
```diff
@@ -24,23 +24,43 @@ The objective is to compare the contributivity figures obtained with the differe
...
+A coalition game is a game where players form coalitions and each coalitions gets a score according to some rules. The winners are the players who manage to be in the coalition with the best score. Here we can consider each data provider is a player, and that forming a coalition is building a federated model using the dataset of each player within the coalition. The score of a coalition is then the performance on a test set of the federated model built by the coalition.<br/><br/>
+To attributes a part of the global score to each player/data providers, we can use the Shapley values. to define the Shapley value we first have to define the "increment" in performance of a player in a coalition. Such "increment"  is the performance of the coalition minus the performance of the coalition without this player. The Shapley value of a player is a properly weighted average of its "increments" in every possible coalition. <br/><br/>
```
Comment (natct10) [inline README.md:40]: typo: **T**o define the Shapley value

File: README.md [hunk 3]
```diff
@@ -24,23 +24,43 @@ The objective is to compare the contributivity figures obtained with the differe
...
+The computation of the Shapley Values quickly becomes intensive when the number of players increases. Indeed to compute the increment of a coalition, we need to fit two federated model, and we need to do this for every possible coalitions. If *N* is the number of players we have to do *2^N* fits to compute the shapley values of each players. As this is quickly too costly, we are considering estimating the shapley values rather then computing it exactly. The estimation metohds considered are :
+        * [done] The exact Shapley Values computation
+Given the limited number of data partners we consider at that stage it is possible to actually compute the Shapley Values with a reasonable amount of resources. 
+        * [done] Monte-Carlo Shapley approximation
+As the sahpley value is an average we can estimate it using the Monte-Carlo method. Here it consists in sampling a raesonable number of increments (says a hundred per player) and to take the  average of the sampled increments of a player as the estimation of the Shapley value of that player.
```
[the remark this answers is not in the document]
Comment (natct10) [reply README.md:45]: typo: raesonable

[the remark this answers is not in the document]
Comment (natct10) [reply README.md:45]: typo: double space between "the" and "average "

File: README.md [hunk 4]
```diff
@@ -24,23 +24,43 @@ The objective is to compare the contributivity figures obtained with the differe
...
+        * [done] Importance sampling methods
+Importance sampling is a method to reduce the number of sampled increments in the Monte-Carlo method while keeping the same accuracy. It consists in sampling the increments according to non-uniform distribution, giving more chance for big increment  than for small increment to be sampled. The  bias induced by altering the sampling distribution is canceled by properly weighting each sample: If an increment is sampled with *X* times more chances, then we weight it by *1/X*. Note that this require to know the value of increment before computing them, so in practice we try to guess the value of the increment. We inflate, resp. deflate, the probability of sampling an increment if we guess its value is big, resp. small.
+     We designed three ways to guess the value of increments, which lead to three different importance sampling methods: 
+            * [done] Linear importance sampling
+            * [done] Regression importance sampling
+            * [done] Adaptative kriging importance sampling   
+         * [done] Stratified Monte Carlo Shapley
```
Comment (natct10) [inline README.md:56]: indentation pb?

File: README.md [hunk 5]
```diff
@@ -24,23 +24,43 @@ The objective is to compare the contributivity figures obtained with the differe
...
+     We designed three ways to guess the value of increments, which lead to three different importance sampling methods: 
+            * [done] Linear importance sampling
+            * [done] Regression importance sampling
+            * [done] Adaptative kriging importance sampling   
+         * [done] Stratified Monte Carlo Shapley  
+"Stratification and with proper allocation" is an other method to reduce the number of sampled increments in the Monte-Carlo method while keeping the same accuracy. There are two ideas behind this method: 1) the Sapley value is a mean of means taken on strata of increments. A strata of increments corresponds the all the increments of coalition with the same number of players. We can estimate the  means on each stata independently rather than the whole mean, this improve the accuracy and reduces the number of increments to sample.  2) We can allocate a different amount of sampled increment to each mean of a strata. If we allocate more sample to the stratas where the increments value varies more, we can reduce the accuracy even more.
```
[the remark this answers is not in the document]
Comment (natct10) [reply README.md:57]: typo: "an other" => another

[the remark this answers is not in the document]
Comment (natct10) [reply README.md:57]: double space: "We can estimate the [here] means"

[the remark this answers is not in the document]
Comment (natct10) [reply README.md:57]: typo: "this improve**S**"

File: README.md [hunk 6]
```diff
@@ -24,23 +24,43 @@ The objective is to compare the contributivity figures obtained with the differe
...
+    As we can estimate the mean of a strata by sampling with replacment of witout replacment, it gives two approximation methods:
```
[the remark this answers is not in the document]
Comment (natct10) [reply README.md:58]: typo: witout => without

**Comment by natct10** [comment]
Great work, crystal clear, thank you @Thomas-Galtier

**Comment by RomainGoussault** [comment]
image
I think we should add some `:` or something in between the method name and the method description. It looks weird like that.

Note this applies to every method name, not just the two that I showed

**Pull request merged**